test/react_lazy_test.dart

```diff
@@ -0,0 +1,26 @@
+@TestOn('browser')
+library react.react_lazy_test;
+
+import 'package:react/react.dart' as react;
+import 'package:react/react_client/react_interop.dart';
+import 'package:test/test.dart';
+
+import 'factory/common_factory_tests.dart';
+
+main() {
+  group('lazy', () {
+    group('- common factory behavior -', () {
+      final LazyTest = react.lazy(() async => react.registerFunctionComponent((props) {
+            props['onDartRender']?.call(props);
+            return react.div({...props});
+          }));
```

Comment:
Can we also add some test coverage for lazy-wrapping a JS component (since it doesn't have the same factory proxy)?

Reply:
I started spiked out what that would look like locally, since I wanted to be more specific than just "add more coverage please" lol, and wanted to make sure it was a reasonable ask if things got hairy.

In doing so, I realized there were some things like event and ref wrapping that weren't behaving correctly, which was a bit concerning. 

I started debugging and eventually realized that we could just add a wrapper in the react-dart layer, like you did [in over_react](https://github.com/Workiva/over_react/pull/941/files#diff-2a180371d85a0d65907ef08aee9aace73e38275f26b90e14fd457a7a0b8533d5R84)!

That fixed, like, all of the test failures I was seeing.

See this PR here with my proposed added tests, and the wrapper in react-dart: https://github.com/Workiva/react-dart/pull/407

Thoughts on putting that wrapper here instead? Without it, react-dart's `lazy` doesn't really work with JS components. I think it makes sense conceptually too, so that the concerns of the interop details stay in react-dart instead of bleeding into react-dart. And we'll get to run all of these additional shared react-dart tests on the implementations, too.

Reply:
makes way more sense to put it here, does that mean we can remove it from the over_react impl?

Reply:
Yeah, that should make the over_react one unnecessary, and I think we should remove it if possible